### PR TITLE
Add documentation examples for `non-iterator-returned`

### PR DESCRIPTION
**Pull request by julthep**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Add documentation examples for `non-iterator-returned`

Refs #5953

### CODE REVIEW

File: doc/data/messages/n/non-iterator-returned/bad.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Iterator:
+    def __init__(self, end, start=0):
+        self.n = start
+        self.end = end
+
+    def __iter__(self):  # [non-iterator-returned]
+        return self
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/n/non-iterator-returned/bad.py:7]:
```suggestion
import random


class GenericAstrology:
    def __init__(self, signs, predictions):
        self.signs = signs
        self.predictions = predictions

    def __iter__(self):  # [non-iterator-returned]
        self.index = 0
        self.number_of_prediction = len(self.predictions)
        return self


SIGNS = ["Aries", "Taurus", "Gemini", "Cancer", "Leo", "Virgo", "Libra"]
PREDICTIONS = ["good things", "bad thing", "existential dread"]
for sign, prediction in GenericAstrology(SIGNS, PREDICTIONS):
    print(f"{sign} : {prediction} today")

```

File: doc/data/messages/n/non-iterator-returned/good.py [hunk 2]
```diff
@@ -1 +1,15 @@
...
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.n <= self.end:
+            n = self.n
+            self.n += 1
+            return n
+
+        raise StopIteration
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/n/non-iterator-returned/good.py:15]:
```suggestion
import random


class GenericAstrology:
    def __init__(self, signs, predictions):
        self.signs = signs
        self.predictions = predictions

    def __iter__(self):
        self.index = 0
        self.number_of_prediction = len(self.predictions)
        return self

    def __next__(self):
        if self.index == len(self.signs):
            raise StopIteration
        self.index += 1
        prediction_index = random.randint(0, self.number_of_prediction - 1)
        return self.signs[self.index - 1], self.predictions[prediction_index]


SIGNS = ["Aries", "Taurus", "Gemini", "Cancer", "Leo", "Virgo", "Libra"]
PREDICTIONS = ["good things", "bad thing", "existential dread"]
for sign, prediction in GenericAstrology(SIGNS, PREDICTIONS):
    print(f"{sign} : {prediction} today")
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Sorry I changed the code a little so the example made a little more sense, simply renaming was not easy. Let me know what you think :)

File: doc/data/messages/n/non-iterator-returned/bad.py (hunk 1 above)
Comment (julthep) [reply doc/data/messages/n/non-iterator-returned/bad.py]: Wow, great "rewrite" :)

File: doc/data/messages/n/non-iterator-returned/good.py (hunk 2 above)
Comment (julthep) [reply doc/data/messages/n/non-iterator-returned/good.py]: Wow, great "rewrite" :)

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**